source/_integrations/homekit_controller.markdown

```diff
@@ -45,67 +45,80 @@ ha_integration_type: integration
 
 The [HomeKit](https://developer.apple.com/apple-home/) controller integration allows you to connect accessories with the "Works with HomeKit" logo to Home Assistant. This integration should not be confused with the [HomeKit](/integrations/homekit/) integration, which allows you to control Home Assistant devices via HomeKit.
 
-The integration will automatically detect HomeKit compatible devices that are ready to pair if the [`zeroconf`](/integrations/zeroconf/) integration is enabled. This is enabled by default on new installations via the [`default_config`](/integrations/default_config/) integration.
+# Adding a HomeKit device
 
-To see which devices have been discovered see the "Integrations" page in your Home Assistant dashboard. When you click on "Configure" you can enter your HomeKit PIN and the device should be added to your Home Assistant instance. If your device is currently paired with an Apple device via HomeKit, you will need to reset it in order to pair it with Home Assistant. Once Home Assistant is configured to work with the device, you can export it back to Siri and Apple Home with the [`HomeKit`](/integrations/homekit/) integration.
+There are different methods to add a HomeKit device to Home Assistant:
 
-## Supported devices
+- [via Ethernet or Wi-Fi](#adding-a-homekit-device-via-ethernet-or-wi-fi): Requires the pairing code
+- [via Bluetooth](#adding-a-homekit-device-through-bluetooth): Requires the pairing code
+- [onto a Thread network via Home Assistant](#adding-a-homekit-device-to-a-thread-network-via-home-assistant): No pairing code required
+- [onto a Thread network via Apple Thread border router](#adding-a-homekit-device-via-apple-thread-border-router): No pairing code required
```

Comment:
I'd indent these to make them sub-options of the Bluetooth option (since the Bluetooth on-boarding is a prerequisite).

By design, all methods in some way or another need the pairing code (e.g. both Thread methods just require that pairing code earlier, when adding to the Bluetooth). Otherwise the whole security concept of these devices would be broken :cold_sweat: . So, I'd just remove that information.

```suggestion
- [via Ethernet or Wi-Fi](#adding-a-homekit-device-via-ethernet-or-wi-fi)
- [via Bluetooth](#adding-a-homekit-device-through-bluetooth)
  - [migrate to a Thread network via Home Assistant](#adding-a-homekit-device-to-a-thread-network-via-home-assistant)
  - [migrate to Thread network via Apple Thread border router](#adding-a-homekit-device-via-apple-thread-border-router)
```